### PR TITLE
pioasm: Ada output format

### PR DESCRIPTION
**Pull request by JeremyGrosser**

### CODE REVIEW

File: tools/pioasm/ada_output.cpp [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        }
+    };
+
+    ada_output() : output_format("ada") {}
+
+    std::string get_description() override {
+        return "Ada specification";
+    }
+
+    FILE *open_spec_output(std::string package_name) {
```
Comment (kilograham) [inline tools/pioasm/ada_output.cpp:25]: this doesn't seem to be used

File: tools/pioasm/ada_output.cpp [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                std::cerr << "error: too many options for Ada format" << std::endl;
+                return 1;
+        }
+
+        FILE *out = open_single_output(destination);
+        if (!out) return 1;
+
+        header(out, "This file is autogenerated by pioasm; do not edit!", 0);
+        fprintf(out, "pragma Style_Checks (Off);\n\n");
+        fprintf(out, "with RP.PIO;\n\n");
```
Comment (kilograham) [inline tools/pioasm/ada_output.cpp:104]: where would users get this from?

Comment (JeremyGrosser) [reply tools/pioasm/ada_output.cpp:104]: https://github.com/JeremyGrosser/rp2040_hal/blob/master/src/drivers/rp-pio.ads

PIO example project here: https://github.com/JeremyGrosser/pico_examples/tree/master/pio_blink

We're still working out the details of all the APIs and build process, so the project is light on documentation at the moment. I expect that situation to improve soon.

Comment (kilograham) [reply tools/pioasm/ada_output.cpp:95]: cool; i guess you are the first there, you can claim "ada', and therefore what it does is up to you... we don't need to have the supporting files, though perhaps a link somewhere would be helpful

Comment (JeremyGrosser) [reply tools/pioasm/ada_output.cpp:102]: I put links to the relevant github repositories in a comment at the top of ada_output.cpp... Is that where it belongs or is there a better place to document this dependency?

Comment (kilograham) [reply tools/pioasm/ada_output.cpp:102]: i think that is fine

**Review by kilograham — DISMISSED** [review]

**Comment by kilograham** [comment]
can you pls rebase this onto develop

**Comment by JeremyGrosser** [comment]
> can you pls rebase this onto develop

Done.

**Review by kilograham — APPROVED** [review]

**Pull request merged**